backends/arm/test/ops/test_depthwise_conv.py

```diff
@@ -0,0 +1,214 @@
+    out_channels=12,
+    kernel_size=(2, 2),
+    stride=(1, 1),
+    groups=6,
+    padding=0,
+    width=4,
+    height=4,
+    batches=1,
+)
+
+dw_conv2d_3x3_1x3x256x256_gp3_st1 = Conv2d(
+    in_channels=3,
+    out_channels=3,
+    kernel_size=(3, 3),
+    stride=(1, 1),
+    groups=3,
+    padding=0,
+    width=256,
+    height=256,
+    batches=1,
+)
+
+dw_conv2d_3x3_1x4x256x256_gp4_st1 = Conv2d(
+    in_channels=4,
+    out_channels=8,
+    kernel_size=(3, 3),
+    stride=(1, 1),
+    groups=4,
+    padding=0,
+    width=256,
+    height=256,
+    batches=1,
+)
+
+dw_conv2d_3x3_2x8x198x198_gp8_st3 = Conv2d(
+    in_channels=8,
+    out_channels=16,
+    kernel_size=(3, 3),
+    stride=3,
+    groups=8,
+    padding=0,
+    width=198,
+    height=198,
+    batches=2,
+)
+
+dw_conv2d_3x3_1x4x256x256_gp4_nobias = Conv2d(
+    in_channels=4,
+    out_channels=8,
+    kernel_size=(3, 3),
+    stride=1,
+    groups=4,
+    bias=False,
+    width=256,
+    height=256,
+    batches=1,
+)
+
+two_dw_conv2d = Conv2d(
+    nbr_conv=2,
+    width=64,
+    height=64,
+    in_channels=[4, 8],
+    out_channels=[8, 24],
+    kernel_size=[(3, 3), (3, 3)],
+    stride=[1, 1],
+    padding=[0, 0],
+    groups=[4, 8],
+    bias=[True, True],
+    batches=2,
+)
+
+# Shenanigan to get a nicer output when test fails.
+testsuite = [
+    ("2x2_1x6x4x4_gp6_st1", dw_conv2d_2x2_1x6x4x4_gp6_st1),
+    ("3x3_1x3x256x256_gp3_st1", dw_conv2d_3x3_1x3x256x256_gp3_st1),
+    ("3x3_1x4x256x256_gp4_st1", dw_conv2d_3x3_1x4x256x256_gp4_st1),
+    ("3x3_2x8x198x198_gp8_st3", dw_conv2d_3x3_2x8x198x198_gp8_st3),
+    ("3x3_1x4x256x256_gp4_nobias", dw_conv2d_3x3_1x4x256x256_gp4_nobias),
+    ("two_dw_conv2d", two_dw_conv2d),
+]
+
+# Expected fails on Ethos-U55/U65. This is a known limitation.
+# Check: https://review.mlplatform.org/plugins/gitiles/ml/ethos-u/ethos-u-vela/+/refs/heads/main/SUPPORTED_OPS.md
+#   For depth multipliers > 1, IFM channels must be 1 and OFM channels must be
+#   equal to the depth multiplier
+# and
+#   depthwise_multiplier = out_channels / in_channels
+testsuite_u55 = testsuite.copy()
+testsuite_u55.remove(("2x2_1x6x4x4_gp6_st1", dw_conv2d_2x2_1x6x4x4_gp6_st1))
+testsuite_u55.remove(("3x3_1x4x256x256_gp4_st1", dw_conv2d_3x3_1x4x256x256_gp4_st1))
+testsuite_u55.remove(("3x3_2x8x198x198_gp8_st3", dw_conv2d_3x3_2x8x198x198_gp8_st3))
+testsuite_u55.remove(
+    ("3x3_1x4x256x256_gp4_nobias", dw_conv2d_3x3_1x4x256x256_gp4_nobias)
+)
+testsuite_u55.remove(("two_dw_conv2d", two_dw_conv2d))
+
+
+class TestDepthwiseConv2D(unittest.TestCase):
+    def _test_dw_conv2d_tosa_MI_pipeline(
+        self, module: torch.nn.Module, test_data: Tuple[torch.Tensor]
+    ):
+        tester = (
+            ArmTester(
+                module,
+                inputs=test_data,
+                profile=TosaProfile.MI,
+                backend=ArmBackendSelector.TOSA,
+                permute_memory_to_nhwc=True,
+            )
+            .export()
+            .to_edge()
```

Comment:
check for conv nodes?

Reply:
I thought about doing that, but it got a bit convoluted with the multi-conv-op tests. That and the fact that the main purpose of the test is to ensure the stages from partition and onwards are what is unique to this Arm backend unit test, made me go without that check. I can add it if you have a string opinion about that. Let me know.